components/service/glean/scripts/sdk_generator.gradle

```diff
@@ -0,0 +1,71 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import org.apache.tools.ant.taskdefs.condition.Os
+
+// Configure the Python environments
+envs {
+    bootstrapDirectory = new File(buildDir, 'bootstrap')
+    envsDirectory = new File(buildDir, 'envs')
+
+    // Setup a miniconda environment.
+    conda "Miniconda3", "Miniconda3-latest", "64"
+}
+
+// Setup the python environment before the build starts.
+
+task createBuildDir {
+    description = "Make sure the build dir exists before creating the Python Environments"
+    onlyIf {
+        !file(buildDir).exists()
+    }
+    doLast {
+        println "Creating build directory:" + buildDir.getPath()
+        buildDir.mkdir()
+    }
+}
+
+// Make sure that a build directory exists before attempting to download
+// the Python environment in the preBuild step. This will fail if the directory
+// doesn't exist.
+preBuild.mustRunAfter(createBuildDir)
+preBuild.dependsOn("build_envs")
+
+// Generate the Metrics API
+android.applicationVariants.all { variant ->
+    def sourceOutputDir =  "$buildDir/telemetry/src/${variant.name}/kotlin"
+    def generateKotlinAPI = task("generateMetricsSourceFor${variant.name.capitalize()}", type: Exec) {
+        description = "Generate the Kotlin code for the Metrics API"
+
+        workingDir  rootDir
+        // Note that the command line is OS dependant: on linux/mac is Miniconda3/bin/python.
+        if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+            commandLine "$buildDir/bootstrap/Miniconda3/python"
+        } else {
+            commandLine "$buildDir/bootstrap/Miniconda3/bin/python"
+        }
+        // TODO use the published glean_parsers package once bug 1497800 lands
+        // The path to the parser needs to be hardcoded now that we're using a stubbed parser.
+        args "$rootDir/components/service/glean/scripts/parser.py"
```

Comment:
Add `@InputFile` definitions for these, so that task chaining works.  Do you know the exact outputs?  Use `@OutputFile`, too.  If not, `@OutputDirectory`.

Reply:
Unfortunately I don't know the exact outputs: we're going to generate multiple files in the near future, not just one. I noticed that these annotations don't work within `tasks`: as far as I understood I would need to generate a new task type by creating a class that inherits from `Exec`, then I could tag the fields of that class using the annotations. @ncalexan , do you think this is still needed? By removing `mustRunAfter` this now seems to work fine.